public/app/features/alerting/AlertTabCtrl.ts

```diff
@@ -140,18 +140,21 @@ export class AlertTabCtrl {
       name: model.name,
       iconClass: this.getNotificationIcon(model.type),
       isDefault: false,
+      uid: model.uid
     });
-    this.alert.notifications.push({ id: model.id });
+    if (!_.find(this.alert.notifications, { id: model.id})) {
+      this.alert.notifications.push({ uid: model.uid });
+    }
 
     // reset plus button
     this.addNotificationSegment.value = this.uiSegmentSrv.newPlusButton().value;
     this.addNotificationSegment.html = this.uiSegmentSrv.newPlusButton().html;
     this.addNotificationSegment.fake = true;
   }
 
-  removeNotification(index) {
-    this.alert.notifications.splice(index, 1);
-    this.alertNotifications.splice(index, 1);
+  removeNotification(deleteUid) {
+    _.remove(this.alert.notifications, { uid: deleteUid});
+    _.remove(this.alertNotifications, { uid: deleteUid});
   }
```

Comment:
@bergquist 
Here was problem. alert.notifications contains custom channel, but quantity channels = default channels + custom channels. So if we have chain of 1 default and 2 custom channels - amount is 3. When you want to delete second custom channel, it deletes 1st custom channel.